dotnet/src/SemanticKernel.Skills/Skills.Compression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Skills.Compression;
+
+//**********************************************************************************************************************
+// EXAMPLE USAGE
+//
+// ISemanticKernel kernel = SemanticKernel.Build();
+// var zipCompressor = new ZipFileCompressor();
+// var skill = new FileCompressionSkill(zipCompressor);
+// var fileCompression = kernel.ImportSkill(skill, "FileCompression");
+// string sourceFilePath = "FileToCompress.txt";
+// string destinationFilePath = "CompressedFile.zip";
+// var variables = new ContextVariables(sourceFilePath);
+// variables.Set(FileCompressionSkill.Parameters.DestinationFilePath, destinationFilePath);
+// await kernel.RunAsync(variables, fileCompression["CompressFileAsync"]);
+//**********************************************************************************************************************
+
+/// <summary>
+/// Skill for compressing and decompressing files.
+/// </summary>
+public class FileCompressionSkill
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Directory to which to extract compressed file's data.
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// File path where to save compressed data.
+        /// </summary>
+        public const string DestinationFilePath = "destinationFilePath";
+    }
+
+    private readonly IFileCompressor _fileCompressor;
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressFileAsync(string sourceFilePath, SKContext context)
```

Comment:
would be nice to return the path of the file created, it could be useful, without having to search the context variables

Reply:
Done